### ISSUE TITLE
Hello World tutorial doesn't work, it keeps loading or checking job is not paused

### ISSUE THREAD

**Issue by brightzheng100** [concourse/concourse#764]
* Concourse version: 2.4.0
* Deployment type (BOSH/Docker/binary): Docker
* Browser (if applicable): Chrome Version 54.0.2840.71 m

Docker 
```
$ docker --version
Docker version 1.12.3, build 6b644ec
```

Docker Compose
```
$ docker-compose --version
docker-compose version 1.6.2, build 4d72027
```

Fly
```
$ fly --version
2.4.0
```

I followed exactly the Docker deployment type to setup my environment.
After I span up the environment I followed the hello world tutorial:
```
$ fly -t lite set-pipeline -p hello-world -c hello-world.yml
$ fly -t lite unpause-pipeline -p hello-world
```

I could see the pipeline was there running then I manually triggered it by clicking the + button.
Then the screen kept hanging here:
```
(tick) checking pipeline is not paused
(running sign, that circle) checking job is not paused
(tick) waiting for a suitable set of input versions
(tick) checking max-in-flight is not reached
```

When I triggered the second time by clicking the + button.
A new tab appeared but the content displayed was: loading
Nothing else.

What should I do?

**Comment by vito** [comment]
are you sure you didn't accidentally pause the job in the web ui? click the
job name from the build page and then click the play button if so.

**Comment by brightzheng100** [comment]
Actually I tried many times. I even restarted the whole environment to ensure I had all this refreshed.
BTW: I'm using Ubuntu 16 Xenial in VirtualBox 5.0.28.

**Comment by brightzheng100** [comment]
```
$ fly -t lite pipelines
name           paused  public
hello-world    no      no
```

Now it keeps pending at the job say-hello.
I'm expecting the result of "Hello, world!" to be printed out...

**Comment by brightzheng100** [comment]
By seeing below output after a long wait time, I eventually understand it's because of proxy issue.
```
resource script '/opt/resource/check []' failed: exit status 1

stderr:
failed to ping registry: 2 error(s) occurred:

* ping https: Get https://registry-1.docker.io/v2/: dial tcp: i/o timeout
* ping http: Get http://registry-1.docker.io/v2/: dial tcp: i/o timeout
```

I have setup below proxy info, what else I need to set?

$ sudo vim /etc/default/docker
```
export http_proxy="http://192.168.56.1:3128/"
export https_proxy="http://192.168.56.1:3128/"
export no_proxy="localhost,127.0.0.1,192.168.56.*"
```

$ vim /etc/systemd/system/docker.service.d/http-proxy.conf
```
[Service]
Environment="HTTP_PROXY=http://192.168.56.1:3128/" "HTTPS_PROXY=http://192.168.56.1:3128/" "NO_PROXY=localhost,127.0.0.1,192.168.56.*"
```

**Comment by ppaulweber** [comment]
@brightzheng100: I've discovered the same problem with my concourse installation. If I click on a job in the web UI and the details view is loaded, the loading of the log never stops and does not show any output. 
But, by accident, I've discovered that if the `concourse web` instance stops (program abort etc.) the log is shown.
@vito: could it be that the web UI correctly fetches the build log, but the display logic does not recognize it?
FYI: I'm running a linux build version `2.4.0` of concourse on a `Ubuntu 16.04 LTS`, `4.4.0-47`, `x86_64`

**Comment by vito** [comment]
@brightzheng100 Proxies should be configured in the worker's env, not Docker's (Concourse isn't talking to that Docker).

**Comment by colthreepv** [comment]
this is a docker-related dns resolve issue, I solved it passing tons of env variables, https://gist.github.com/colthreepv/6b818cfcf296dc1b5c2cf15eb76a140e

**Comment by brightzheng100** [comment]
Thanks @vito and @colthreepv for your advice.

By following the direction, I tried another way instead and it works also, like below:
```
concourse-worker:
  image: concourse/concourse
  privileged: true
  links: [concourse-web]
  command: worker
  volumes: ["./keys/worker:/concourse-keys"]
  environment:
    CONCOURSE_TSA_HOST: concourse-web
    **http_proxy: http://192.168.56.1:3128
    https_proxy: http://192.168.56.1:3128
    no_proxy: localhost,127.0.0.1,192.168.56.119**
```

I suggest to make this documented explicitly to avoid any disappointment for newbies.

**Comment by vito** [comment]
Glad to see you figured it out! I don't feel great about adding that to the docs though as we just conform to the standard configuration of proxies. They are also listed in the `--help` text for the binaries, which the docs point to.

**Comment by DanPearson** [comment]
I got this issue as well running the binary image but found another root cause.   Eventually found that it's caused by Docker and IPv6 on the host.   A simple and quick fix is to disable IPv6 and restart the Web and Worker.   I found the following messages in the syslog;

docker0: port 1(veth39c7847) entered disabled state
BTRFS: could not find root 8
IPv6: ADDRCONF(NETDEV_UP): wbrdg-0afe0008: link is not ready

Hope this helps someone else down the line.
